### PR TITLE
Bach[Athena]: Support SeriesNumericInterval, Cut and QCut Operation

### PR DESCRIPTION
**Pull request by KathiaBarahona**

### CODE REVIEW

File: bach/bach/series/series_multi_level.py [hunk 1]
```diff
@@ -455,18 +463,32 @@ def to_pandas_info(self) -> Optional[ToPandasInfo]:
...
+            return ToPandasInfo(dtype='object', function=self._parse_numeric_interval_value_athena)
         return None
 
     def get_column_expression(self, table_alias: str = None) -> Expression:
         # construct final column based on levels
         if is_postgres(self.engine):
             # casting is needed since numrange does not support float64
-            base_expr_stmt = f'numrange(cast({{}} as numeric), cast({{}} as numeric), {{}})'
+            # we cast first to text, as casting double precision to numeric might round decimal values
+            base_expr_stmt = f'numrange({{}}::text::numeric, {{}}::text::numeric, {{}})'
```
Comment (KathiaBarahona) [inline bach/bach/series/series_multi_level.py:475]: Not sure why the rounding happens tho :/ parsing to text and then to numeric worked. If there is a better solution, please let me know

File: bach/tests/functional/bach/test_df_groupby.py [hunk 2]
```diff
@@ -718,18 +717,10 @@ def test_groupby_w_multi_level_series(engine):
...
-        range_3 = NumericRange(lower=Decimal('2.'), upper=Decimal('5.'), bounds='[]')
-        range_4 = NumericRange(lower=Decimal('3.'), upper=Decimal('7.'), bounds='[]')
-    elif is_bigquery(engine):
-        range_1 = {'lower': 0., 'upper': 1., 'bounds': '[]'}
-        range_2 = {'lower': 1., 'upper': 3., 'bounds': '[]'}
-        range_3 = {'lower': 2., 'upper': 5., 'bounds': '[]'}
-        range_4 = {'lower': 3., 'upper': 7., 'bounds': '[]'}
-    else:
-        raise Exception()
+    range_1 = pd.Interval(0., 1., closed='both')
```
Comment (thijs-obj) [inline bach/tests/functional/bach/test_df_groupby.py:720]: much nicer :+1:

File: bach/bach/series/series_multi_level.py [hunk 3]
```diff
@@ -408,7 +415,8 @@ class SeriesNumericInterval(SeriesAbstractMultiLevel):
     dtype_aliases = (pandas.Interval, 'numrange')
     supported_db_dtype = {
         DBDialect.POSTGRES: 'numrange',
-        DBDialect.BIGQUERY: 'STRUCT<lower FLOAT64, upper FLOAT64, bounds STRING(2)>'
+        DBDialect.BIGQUERY: 'STRUCT<lower FLOAT64, upper FLOAT64, bounds STRING(2)>',
+        DBDialect.ATHENA: 'STRUCT<lower: double, upper: double, bounds: varchar(2)>',
```
Comment (thijs-obj) [inline bach/bach/series/series_multi_level.py:419]: This doesn't matter too much, but is this correct? Later on we cast this thing to json?

**Review by thijs-obj — APPROVED** [review]
small comment. Looks otherwise good to me

File: bach/bach/series/series_multi_level.py (hunk 3 above)
Comment (KathiaBarahona) [reply bach/bach/series/series_multi_level.py:419]: TBH the values on that mapping are just a hack for this series to actually work with the type registry and to add support for the engine without making any big changes.

Comment (thijs-obj) [reply bach/bach/series/series_multi_level.py:419]: Just `DBDialect.ATHENA: None,` would work, wouldn't it?

**Pull request merged**